### ISSUE TITLE
Misc repair kit wastes duct tape when used for anyting other than repairing

### ISSUE THREAD

**Issue by martosss** [CleverRaven/Cataclysm-DDA#55158]
### Describe the bug

Any action with the misc repair kit costs Duct Tape!
```diff 
+ 1) repair items - new functionality                                                  costs duct tape
- 2) use any of the X-Acto / wood saw actions - cut an item, chop logs, chop trees ... costs duct tape
```

Fun fact - you can disassemble the misc repair kit, then use the X-Acto knife/wood saw for free, then reassemble the kit ... which is stupid.
```diff
- no extra realism added
- extra inventory cost to store the misc repair kit as single unit rather than its components as separate items
- wastes duct tape - what rainbows are made of!
```

### Steps to reproduce

1) spawn 4 logs, an X-Acto knife and a wood saw
```diff
+2) cut a log using a simple wood saw - free
+3) cut a log using an X-Acto "cut up item" option - free

-4) cut a log using a misc repair kit's "chop log" option - costs 5 duct tape
-5) cut a log using a misc repair kit's "cut up item" option - costs 5 duct tape 
```

### Expected behavior

**X-Acto knife** and **wood saw** usage costs **no duct tape** - no matter if you use them from the misc repair kit or by themselves.

### Versions and configuration

win 8.1, tested both on Stable 0.F-3 and Experimental branch from 22.01.22, Tiles, English, default mods

**Comment by chrispikula** [comment]
See
#54185
#55123


I thought this was fixed...?  Oh, it was.

#30137

Appears that the root issue is

#54120

Associated:

#55123

Needless to say, this has been an ongoing thorny issue.

**Comment by martosss** [comment]
Note that those are related to the "firearm repair kit", and not the "misc repair kit".
I tested on a 10-day old experimental and the issue for the misc repair kit is still there?

**Comment by chrispikula** [comment]
It's still part of the use_actions issue, I think.  Charges getting used in a multi-purpose item, even when that particular sub-use shouldn't use charges.

**Comment by martosss** [comment]
Yes, I agreethey are related, but I don't think this issue in particular is fixed, yet? I think the "solution" is currently only applied to the firearm repair kit, but it needs to be extended to other types of kits?

**Comment by Maleclypse** [comment]
I’ve labeled this good first issue for someone to replicate the firearm kit fix to other items. A better fix would be making it configurable in json as to what qualities/uses need charges.

**Comment by wapcaplet** [comment]
Items already have a "charged_qualities" field that is supposed to do this (#52304) but I don't know if those work correctly after the introduction of "ammo_scale" (#52304).

**Comment by a-chancey** [comment]
I'm running into a separate issue with the kit now, but I need to see if it's still present on current experimental. 

At fab 6, my |. fire axe which dropped from a feral says it has a 9% success/0% fail, and if I repair until success/failure/level up, it just burns all 200 duct tape in the kit and says it doesn't have enough charges to continue. 

Either the success rate is displaying wrong (possible, fire axes need fab 9 to craft), or something else is going on with it just not repairing. 

I'll try to check it out with the debugger running later if I get a chance

**Comment by chrispikula** [comment]
It depends on how many charges it uses per attempt, might be 10 or 15, I don't recall.  Simplifying the probabilities to 1 in 10, w/ 15 charges per use, and 13 attempts in 200 charges, you'll see that it's about [1/4](https://anydice.com/program/26c8f) that you'll fail before you run out of charges.  Actually, a little bit worse, because it's 9%, and not 10%, but that's not going to make a huge change.


If there's added penalties from other sources, it might worse, but I don't know if those exist.

**Comment by RoyBerube** [comment]
This affects more than just the misc repair kit.

I see that the kit itself is being passed as the tool item to the activity actor, and then has charges deducted in some way by the actor. Not all activities work the same way - cutting does not alter the charges but returns the amount to be deducted.

Cutting - salvage_actor() - returns a value, while prying - prying_activity_actor() - deducts charges directly from the item every do_turn(). We need a solution that does not require rewriting these activities.

I'm thinking that it would be simplest to find a way to pass the actual tool to the activity. For example when prying, pass a crowbar to the activity instead of the firearm repair kit. Then the kit does have any charges deducted.

I can imagine two ways to approach this: create a pseudo tool as required, or include all the tools in the kit. To include the tools in the kit would require some reworking of how kits are implemented, and is probably more work than the pseudo approach.
